plugins/rocketmq/src/main/java/com/navercorp/pinpoint/plugin/rocketmq/interceptor/ProducerSendInterceptor.java

```diff
@@ -0,0 +1,306 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.navercorp.pinpoint.plugin.rocketmq.interceptor;
+
+import static com.navercorp.pinpoint.bootstrap.sampler.SamplingFlagUtils.SAMPLING_RATE_FALSE;
+import static org.apache.rocketmq.common.message.MessageDecoder.NAME_VALUE_SEPARATOR;
+import static org.apache.rocketmq.common.message.MessageDecoder.PROPERTY_SEPARATOR;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.rocketmq.common.protocol.header.SendMessageRequestHeader;
+
+import com.navercorp.pinpoint.bootstrap.async.AsyncContextAccessor;
+import com.navercorp.pinpoint.bootstrap.context.AsyncContext;
+import com.navercorp.pinpoint.bootstrap.context.Header;
+import com.navercorp.pinpoint.bootstrap.context.MethodDescriptor;
+import com.navercorp.pinpoint.bootstrap.context.SpanEventRecorder;
+import com.navercorp.pinpoint.bootstrap.context.Trace;
+import com.navercorp.pinpoint.bootstrap.context.TraceContext;
+import com.navercorp.pinpoint.bootstrap.context.TraceId;
+import com.navercorp.pinpoint.bootstrap.context.scope.TraceScope;
+import com.navercorp.pinpoint.bootstrap.interceptor.AroundInterceptor;
+import com.navercorp.pinpoint.bootstrap.logging.PLogger;
+import com.navercorp.pinpoint.bootstrap.logging.PLoggerFactory;
+import com.navercorp.pinpoint.plugin.rocketmq.RocketMQConstants;
+import com.navercorp.pinpoint.plugin.rocketmq.field.accessor.EndPointFieldAccessor;
+
+/**
+ * @author messi-gao
+ */
+public class ProducerSendInterceptor implements AroundInterceptor {
+    private final PLogger logger = PLoggerFactory.getLogger(getClass());
+    private final boolean isDebug = logger.isDebugEnabled();
+    private static final String SCOPE_NAME = "ROCKETMQ_ASYNC_TRACE_SCOPE";
+    private final MethodDescriptor methodDescriptor;
+    private final TraceContext traceContext;
+
+    public ProducerSendInterceptor(MethodDescriptor methodDescriptor, TraceContext traceContext) {
+        this.methodDescriptor = methodDescriptor;
+        this.traceContext = traceContext;
+    }
+
+    @Override
+    public void before(Object target, Object[] args) {
+        if (isDebug) {
+            logger.beforeInterceptor(target, args);
+        }
+        try {
+            Trace trace = traceContext.currentTraceObject();
+            final Object sendCallback = getSendCallback(args);
+            // async send process
+            if (sendCallback instanceof AsyncContextAccessor) {
+                if (isSkipTrace()) {
+                    return;
+                }
+                final AsyncContext asyncContext =
+                        ((AsyncContextAccessor) sendCallback)._$PINPOINT$_getAsyncContext();
+                ((AsyncContextAccessor) target)._$PINPOINT$_setAsyncContext(asyncContext);
```

Comment:
I have been wondering where to use this.
I could not find place that used it.

Reply:
You are right,I 'll remove it.

Reply:
> I have been wondering where to use this.
> I could not find place that used it.

You are right,I 'll remove it.

Reply:
I think it worked out well.
If it was deleted, then I think `target.addField(AsyncContextAccessor.class); in MQClientAPIImplTransform` looks to be not necessary. 
Please check it :)

Reply:
fix it.